qiita_pet/handlers/analysis_handlers.py

```diff
@@ -59,72 +59,30 @@ def check_analysis_access(user, analysis):
-                        counts[study][meta][sample[pos+1]] += 1
-                        fullcounts[meta][sample[pos+1]] += 1
-            # if no samples left, remove the study from results
-            if len(topop) == len(samples):
-                studypop.append(study)
-                continue
-            # remove already selected samples
-            topop.sort(reverse=True)
-            for pos in topop:
-                samples.pop(pos)
-        for study in studypop:
-            results.pop(study)
-        return results, counts, fullcounts
-
     def _selected_parser(self, analysis):
-        """builds dictionaries of selected samples from analysis object"""
-        selsamples = {}
-        selproc_data = defaultdict(list)
-        for proc_data_id, samps in viewitems(analysis.samples):
-            study = ProcessedData(proc_data_id).study
-            selproc_data[study].append(proc_data_id)
-            selsamples[study] = set(samps)
-        return selproc_data, selsamples
-
-    def _parse_form_select(self):
-        """parses selected checkboxes and yields the selected ones in
-        format accepted by Analysis.add_samples()
-        """
-        # get the selected studies and datatypes for studies
-        for s in self.get_arguments("availstudies"):
-            study_id, proc_data_id = s.split("#", 1)
-            proc_data_id = int(proc_data_id)
-            # get new selected samples for each study and yield with proc id
-            for sample in self.get_arguments(study_id):
-                yield (proc_data_id, sample)
-
-    def _parse_form_deselect(self):
-        """parses selected checkboxes and returns the selected ones in
-        format accepted by Analysis.remove_samples()
-        """
-        studyinfo = self.get_arguments("selstudies")
-        proc_data = []
-        samples = []
-        for sid in studyinfo:
-            # get the processed data ids and add it to the study
-            proc_data.extend(self.get_arguments("dt%s" % sid))
-            # get new selected samples for each study and add to study
-            samples.extend(self.get_arguments("sel%s" % sid))
-        return proc_data, samples
+        """returns the study_proc_id and proc_id_samples for the previously
+           selected samples"""
+        study_proc_data = defaultdict(lambda: defaultdict(list))
+        proc_data_samples = analysis.samples
+        for pid in proc_data_samples:
+            proc_data = ProcessedData(pid)
+            study_proc_data[proc_data.study][proc_data.data_type()].append(pid)
+        return study_proc_data, proc_data_samples
+
+    def _parse_deselect_checkbox_data(self):
+        """splits the list values into proc_data_id and sample_id, then builds
+           dict of results as {proc_data_id: [ samp1, samp2, ...], ...} """
+        data = self.get_arguments('samples-sel')
+        res = defaultdict(list)
+        for val in data:
+            hold = val.split('#', 1)
```

Comment:
IMHO, I think this deserves a comment -- what is the expected format here? I can guess at it, but it would be good to have explicitly stated, probably.